File: src/main/java/ru/diasoft/micro/dofacade/mock/PageflowRes.java
package ru.diasoft.micro.dofacade.mock;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PageflowRes<T> {

    @JsonProperty("instanceId")
    String instanceId;
    @JsonProperty("type")
    String type;
    @JsonProperty("item")
    PageflowItem<T> item;
    @JsonProperty("status")
    String status;
    @JsonProperty("message")
    String message;

}

File: src/main/java/ru/diasoft/micro/dofacade/json/PageflowRes.kt
package ru.diasoft.micro.dofacade.json

import com.fasterxml.jackson.annotation.JsonProperty

class PageflowRes<T>(

        @JsonProperty("instanceId")
        val instanceId: String?,
        @JsonProperty("type")
        val type: String?,
        @JsonProperty("item")
        val item: PageflowItem<T>,
        @JsonProperty("status")
        val status: String?,
        @JsonProperty("message")
        val message: String?
)

class PageflowItem<out T>(

        @JsonProperty("inputParams")
        val params: T,
        @JsonProperty("type")
        val type: String
)
